src/Microsoft.Framework.ConfigurationModel.Json/JsonConfigurationExtension.cs

```diff
@@ -1,12 +1,37 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.IO;
+
 namespace Microsoft.Framework.ConfigurationModel
 {
     public static class JsonConfigurationExtension
     {
         public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path)
         {
+            if (!Path.IsPathRooted(path))
+            {
+                if (configuration.BasePath == null)
+
+                {
+                    throw new InvalidOperationException(
+                        string.Format("The {0} is not rooted and the {1} is null in this {2}",
```

Comment:
I don't like having IConfigurationSourceRoot in the message at all, it isn't a type you need to be familiar with to consume the API and we've deliberately made it possible to ignore it for the most part.

I'd like something more like: $"Unable to resolve relative path '{path}'. You must provide a full path or provide the {nameof(configuration.basePath)} parameter to your Configuration."